.github/workflows/goreleaser.yaml

```diff
@@ -0,0 +1,60 @@
+name: goreleaser
+
+on:
+  push:
+    tags:
+      - '*'
+jobs:
+  goreleaser:
+    environment: arlon
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      -
+        name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.17
+      -
+        name: Cache go modules
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      -
+        name: Test controllers
+        uses: phalanks/envtest-action@v1.0.0-rc
+        with:
+          version: '0.20.2'
+          args: './controllers/...'
+      -
+        name: Run package tests
+        run: |
+          go test -v ./pkg/...
```

Comment:
- In general, avoid `run` logic in GH Action files, instead run a make target, like `make test-local`
- You can make that change later.

Reply:
Sure. BTW, @bcle `make generate` target fails for some reason. The CLI panics. Is this a known issue?

Reply:
Panics? Can you show a log. Discuss on Slack if necessary.

Reply:
`generate` calls `controller-gen` and I think we are using an old version of controller-gen@v0.4.1 - can we try this with controller-gen@v0.7.0 for now , and controller-gen@v0.8.x once we move to k8s 1.23?

Reply:
> * In general, avoid `run` logic in GH Action files, instead run a make target, like `make test-local`
> * You can make that change later.

done